File: PWD001/example-omp.c
// PWD001: Invalid OpenMP multithreading datascoping
// https://www.appentra.com/knowledge/checks/pwd001

void example(int** result, unsigned rows, unsigned cols) {
 int i, j;
 
 // j is implicitly shared and it should be private!
 #pragma omp parallel for shared(result)
 for (i = 0; i < rows; i++) {
   for (j = 0; j < cols; j++) {
     result[i][j] = 0;
   }
 }
}

File: PWD007/example-omp.f90
! PWD007: Unprotected multithreading recurrence
! https://www.appentra.com/knowledge/checks/pwd007

SUBROUTINE foo
    IMPLICIT NONE
    INTEGER :: i, x(5), y(5)

    y(1)=0
    !$omp parallel do
    DO i = 2, 5
        y(i) = y(i-1) + x(i-1)
    END DO
END SUBROUTINE
